include/onnxruntime/core/platform/EigenNonBlockingThreadPool.h

```diff
@@ -133,6 +133,41 @@ namespace concurrency {
 class ThreadPoolParallelSection;
 class ThreadPoolLoop;
 
+/* Usage:
+1. In executor, call Start() before profiling and Stop() to get profiled numbers;
+2. Inside thread pool, call LogStart() before interested section and LogEnd... after to log elapsed time;
+3. To extend, just add more events in enum Event before "All", and update GetEventName(...) accordingly.
+4. Note LogStart must pair with either LogEnd or LogEndAndStart, otherwise ORT_ENFORCE will fail.
+*/
+class ThreadPoolProfiler {
```

Comment:
Negative for now, since EigenNonBlockingThreadPool.h consumes the profiler.
But we do have a plan for refactoring moving several declarations to more proper place.